src/main/java/de/tum/in/www1/artemis/service/ExerciseDateService.java

```diff
@@ -89,6 +101,23 @@ public boolean isAfterLatestDueDate(Exercise exercise) {
         return !isBeforeLatestDueDate(exercise);
     }
 
+    /**
+     * Checks if due date is before now
+     * The due date we use to check depending on whether it is present is individual -> exercise -> none
+     * If no due date is set, returns an empty optional
+     *
+     * @param exercise for which this should be checked.
+     * @return Optional of true, if the due date is before the current time.
+     */
+    public Optional<Boolean> getOptionalIsAfterEarliestDueDate(Exercise exercise) {
+        final ZonedDateTime now = ZonedDateTime.now();
+        return getEarliestIndividualDueDate(exercise).map(now::isAfter);
+    }
```

Comment:
The reasoning behind returning an optional instead of defaulting to false is that returning false does not always allow for the correct control flow (defaulting to false here similar to how it is handled in the methods above (89 - 102)

I am as disappointed as you are about that name, however our dear linter complained about a linguistic smell, as `is...` implies a return of a boolean

Reply:
@simon-hng The linter can be ignored in such cases, it is merely a help for the code review.

Reply:
https://github.com/ls1intum/Artemis/pull/5670/commits/9dfd40a83e4f9e5e02ae41318978248e0d48b91e